operator/v1/types_openshiftcontrollermanager.go

```diff
@@ -0,0 +1,44 @@
+package v1
+
+import (
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+
+// OpenShiftControllerManagerConfig provides information to configure openshift-controller-manager
+type OpenShiftControllerManagerConfig struct {
+	metav1.TypeMeta `json:",inline"`
+}
+
+// +genclient
+// +genclient:nonNamespaced
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+
+// OpenShiftControllerManagerOperatorConfig provides information to configure an operator to manage openshift-controller-manager.
+type OpenShiftControllerManagerOperatorConfig struct {
```

Comment:
`OpenShiftControllerManager` i think

Reply:
See my obsoleted comment:

```
@deads2k i have a feeling you're going to suggest dropping the Config suffix here(since you don't have it on your types), so i'm going to say up front that i think the suffix lends clarity and i'd prefer to keep it for our operator.
```

If it were in a "config" api group i might take a different view, but "openshiftcontrollermanager.operator.openshift.io" is lacking the word config and i consider that a bad thing.

As for whether Operator should be dropped or not...  i'd say I can see both sides but i'm willing to drop it based on the argument that the information in this struct drives the controller manager behavior, not the operator behavior, as well as the fact that operator is in the group name.

Reply:
Operator stutters on the group name, so we chose to drop it.  Config didn't appear to add value since everything in the type and group is documented to manage a thing.  `openshiftcontrollermanagerconfig.operator` doesn't have more qualifying power than `openshiftcontrollermanageroperator`.  Since we know we'll never have more than one and even if we did the config doesn't really clarify compared to the unsuffixed name.

Reply:
it doesn't clarify anything for you because you have all the design+context of the system in your head already and you don't have to simultaneously worry about/interact with the fifty other parts of the system that may use other naming conventions/patterns/etc for a variety of reasons.  

To a user who didn't design and build the system this name says "this is config information for the controller manager" w/o having to read or remember any api docs or have been in the room at the time the api naming conventions/assumptions were established.